pipeline_dp/budget_accounting.py

```diff
@@ -1,20 +1,91 @@
 """Privacy budget accounting for DP pipelines."""
 
 
+class Budget:
+  """Manages the budget allocated for an operation."""
+
+  def __init__(self, weight, use_eps, use_delta, eps=None, delta=None):
+    """Constructs a Budget.
+
+    The values for eps and delta will be computed only after the pipeline was
+    constructed and the method compute_budgets was called.
+
+    Args:
+      weight: The weight is used to get more accurate results.
+      use_eps: A boolean that is False when the operation doesn't need epsilon.
+      use_delta: A boolean that is False when the operation doesn't need delta.
+    """
+    self._eps = eps
+    self._delta = delta
+    self.weight = weight
```

Comment:
Thought again, I think from the API point of view it's better to avoid having weight/use_eps/use_delta in public API of this class. These fields are mostly implementation details.

My suggestions is the following, introduce @ dataclass RequestedBudget (or suggest another name)
with fields
budge:Budget
weight
use_eps
use_delta

And to keep in BudgetAccountant a list of RequestedBudget.

WDYT?

Reply:
It makes a lot of sense since the fields: weight, use_eps, use_delta are not useful outside the methods request_budget and compute_budgets. I will modify the classes. Thanks for pointing it out!